openxla/stablehlo
# Implement GitHub Actions for lint and buildAndTest

Initial implementation of GitHub Actions CI. This changelist implements two actions:

Build & Test:
  - Check out the llvm revision specified in `llvm_version.txt`
  - Build LLVM with ccache enabled (first time cost ~1.5hr, following runs ~10-15min)
  - Build StableHLO with ccache
  - Run `ninja check-stablehlo`
  - Fails if any build or test failures.

Lint:
  - Run `git-clang-format` on the files being submitted.
  - Fails if any files' formatting are updated.

Examples of what the CI page will look like can be seen at: https://github.com/GleasonK/stablehlo/actions

Future work:
- We can consider only running buildAndTest on linux in the future, mac is a bit of a redundancy, but will help us distinguish a real failure from a build configuration issue in GH Actions.
- The lint action worked in the limited testing that I ran on it, but could use a bit more testing. I think that the production repo will be a good way to test if everything is in order.
- Consider any other tests/linting we want to run. If we add python bindings, there is some python formatting and linting that can be added later.
